handlers.go

```diff
@@ -302,8 +302,17 @@ func (t *TricksterHandler) fetchPromQuery(originURL string, params url.Values, r
 		cacheKeyBase += strings.Join(authorization, " ")
 	}
 
-	if t, ok := params[upTime]; ok {
-		end, err = strconv.ParseInt(t[0], 10, 64)
+	if ts, ok := params[upTime]; ok {
+		// time is actually a decimal value to include sub-second granularity.
+		// As such we need to parse the value as a float64. For the sake of the
+		// rest of the code here we'll cast back to an int64 (effectively
+		// truncating the precision)
+		floatEnd, err := strconv.ParseFloat(ts[0], 64)
+		if err != nil {
+			level.Error(t.Logger).Log(lfEvent, "error parsing time", "params", params, lfDetail, err.Error())
+			return []byte{}, &http.Response{}
```

Comment:
Is this PR against an up-to-date `master` branch? The upstream version already allows returning errors from `fetchPromQuery()`, which we should do here too. https://github.com/Comcast/trickster/blob/master/handlers.go#L363